src/generate.rs

```diff
@@ -85,11 +91,26 @@ fn generate_example(expression: &str) -> Result<String, Vec<String>> {
     if let Some(source_types) = components.get(0) {
         let mut sources = IndexMap::new();
 
-        for (i, source_type) in source_types.iter().enumerate() {
-            let name = format!("source{}", i);
+        for (i, source_expr) in source_types.iter().enumerate() {
+            let (name, source_type) = if let Some(c_index) = source_expr.find(':') {
```

Comment:
I feel like this is a situation where a `match` is a bit cleaner, but c'este la vie. :)